python/GafferUI/NumericPlugValueWidget.py

```diff
@@ -84,37 +86,54 @@ def getToolTip( self ) :
 
 		result = GafferUI.PlugValueWidget.getToolTip( self )
 
-		if self.getPlug() is not None :
+		if self.getPlugs() :
 			if result :
 				result += "\n"
 			result += "## Actions\n"
 			result += " - Cursor up/down to increment/decrement\n"
 
 		return result
 
-	def _updateFromPlug( self ) :
+	def _updateFromPlugs( self ) :
+
+		# Get value and error state
 
-		plug = self.getPlug()
-		if plug is not None :
+		value = None
+		errored = False
+		with self.getContext() :
+			try :
+				value = sole( p.getValue() for p in self.getPlugs() )
+			except :
+				errored = True
 
-			with self.getContext() :
-				try :
-					value = plug.getValue()
-				except :
-					value = None
+		with Gaffer.BlockedConnection( self.__valueChangedConnection ) :
 
+			# Always give the widget a value, even if we have multiple, because
+			# the _type_ (int or float) is important, and affects interaction
+			# with the widget.
 			if value is not None :
-				with Gaffer.BlockedConnection( self.__valueChangedConnection ) :
-					self.__numericWidget.setValue( value )
+				self.__numericWidget.setValue( value )
+			elif self.getPlugs() :
+				self.__numericWidget.setValue( next( iter( self.getPlugs() ) ).defaultValue() )
+
+			# But if there are multiple values or an error, clear the actual
+			# display so we don't show anything misleading.
+			if errored or value is None :
+				self.__numericWidget.setText( "" )
+				self.__numericWidget._qtWidget().setPlaceholderText( "---" )
```

Comment:
I don't have a strong opinion beyond it being better to have a placeholder than to have a stale value (which I think is what we had before). I think the garish red background is sufficient to ensure that people won't think the `---` refers to multiple values, and I don't think a `!!!` or `ERROR!" placeholder would add anything but more garishness. I'm thinking of `---` as "indeterminate" rather than "multiple" necessarily. But if you have a bright idea for a nice placeholder I'd be interested to hear it. Or I guess we could just have an empty field?

Reply:
I guess I liked the idea of an empty field as it felt like it was unnecessary busy-ness for widgets that can be empty, and has the potential to keep them similar in appearance to other widgets with that don't support multiple plugs yet. But as we don't have that luxury for bools, lets see how it goes as is...